src/details/ArborX_DetailsLegacy.hpp

```diff
@@ -39,8 +39,8 @@ class LegacyValues
   KOKKOS_FUNCTION
   auto operator()(size_type i) const
   {
-    if constexpr (std::is_same_v<BoundingVolume,
-                                 typename AccessTraitsHelper<Access>::type>)
+    using Primitive = std::decay_t<decltype(Access::get(_primitives, i))>;
```

Comment:
Why not 
`c++
    using Primitive = typename AccessValues<Primitives, PrimitivesTag>::value_type;
`

Reply:
Separation of concerns, keeping dependency minimal.  There is no good reason to bring `AccessValues` into the picture here.  Expand the lines below
https://github.com/arborx/ArborX/blob/9551119fd76fe178683a63592ade6b6c1229d1a4/src/details/ArborX_DetailsLegacy.hpp#L42-L52

Reply:
I'm not sure what you mean about separation of concerns and dependency. How is it different from bringing it to the Callback checks?

Reply:
> I'm not sure what you mean about separation of concerns and dependency.

Software design principle.  Struggle for minimal overlapping between separate functionalities

> How is it different from bringing it to the Callback checks?

The callbacks already had a dependency on access traits.  I would love to get rid of that dependency but it is much more delicate.

Reply:
> The callbacks already had a dependency on access traits.

I'm sorry but I still don't understand. Yes, it did have a dependency. Similarly, this code in Legacy also had a preexisting dependency. What am I missing? I just don't see the difference between the two.